### ISSUE TITLE
Improvement for spaces and ponctuations.

### ISSUE THREAD

**Issue by MaethorNaur** [8VIM/8VIM#345]
**Is your feature request related to a problem? Please describe.**

Currently with the default layouts you have to lift your finger from the keyboard to add a space and a word, this breaks a bit of the flow for me, and also I don't always have a space after a word.

Also, not all languages have the same rules for punctuation. 
For instance, in French, there's a space before and after `?`, whereas in English it's only after. 
Other keyboards handle that situation. And I miss the double tapping on the space button to have a `.`

**Describe the solution you'd like**

- Add movement to add a space without lifting the finger
- Not adding a space when we lift the finger and wrote something (or maybe make it configurable).
- Add spaces accordingly to the language from the layout (updating the layout schema to specify if that layout support that feature and which locale to use). Also, this could be an option (disabled by default).
- Handle double tap on the circle to add `.`

**Describe alternatives you've considered**

Not at the moment

**Additional context**

None

**Comment by MaethorNaur** [comment]
There's already an issue for the double tap for the period: #265

**Comment by mentalisttraceur** [comment]
Yeah I also have often thought over the years ever since 8pen that the ideal swirly input method probably has the space as just another character on the circle (or at least can, by being fully remappable).
